src/Nether.Analytics/OutputFormatters/IHeaderAwareOutputFormatter.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Nether.Analytics
+{
+    /// <summary>
+    /// The interface enables the output formatter to be aware of the header. 
+    /// It also provides certain methods that enable features like serializing the 
+    /// message with the header for certain scenarios (i.e. CSV).
+    /// </summary>
+    public interface IHeaderAwareOutputFormatter : IOutputFormatter
```

Comment:
For future reference, having discussed this with @krist00fer, we've agreed to solve this in a different way, to take into account both views - in a separate PR.